app/controllers/categories_controller.rb

```diff
@@ -1,6 +1,17 @@
 class CategoriesController < ApplicationController
+  include UniqueResourceUrl
+
+
   def show
-    @category = Category.friendly.find(params[:id])
+    @category = category
+    ensure_or_redirect_to(@category)
     @posts = @category.posts.includes(:comments, :categories).paginate(page: params[:page], per_page: 30)
```

Comment:
the idea about having category memoized, is to reuse it. :)
so since you call it anyways, you don't need to assign to `@category` 

`ensure_or_redirect_to(category)` 

and `@posts = category.posts`

if you need `@category` in the view, you either make it a helper method on the controller, or expose the variable be removing the undescore in `@_category` :) 

I prefer the `_` syntax to indicate if it's not used in the view. Didn't know your complete requirement when I reviewed it the last time :)

Reply:
Ah yes, of course. Need to cultivate the habit of thinking things through and not euphorically implement something.